do/sandbox.go

```diff
@@ -79,3 +130,91 @@ func (n *sandboxService) Stream(cmd *exec.Cmd) error {
+	path := "v2/function/namespaces/namespace"
+	body := postBody{Namespace: inputNamespace{Token: token}}
+	req, err := n.client.NewRequest(ctx, http.MethodPost, path, body)
+	if err != nil {
+		return SandboxCredentials{}, err
+	}
+	tokenDecoded := new(responseBody)
+	_, err = n.client.Do(ctx, req, tokenDecoded)
+	if err != nil {
+		return SandboxCredentials{}, err
+	}
+	ans := SandboxCredentials{
+		ApiHost: tokenDecoded.Namespace.ApiHost,
+		Auth:    tokenDecoded.Namespace.Uuid + ":" + tokenDecoded.Namespace.Key,
+	}
+	return ans, nil
+}
+
+// ResolveNamespace resolves a namespace name into a set of credentials for the sandbox.
+// If "" is given as the namespace name, the available namespaces are retrieved and the
+// function attempts to identify one of them as the sandbox namespace.
+// Note: at present, the "" option only works when the customer has exactly one namespace
+// whose 'label' field contains the substring 'sandbox'.   This is subject to change.
+// Note: at present, two remote calls are needed to go from a namespace name to the needed
+// credentials.  First, a JWT token is remotely generated, then ResolveToken call is used
+// to resolve it.  This will be streamlined in the future.
+func (n *sandboxService) ResolveNamespace(ctx context.Context, namespace string) (SandboxCredentials, error) {
+	var err error
+	if namespace == "" {
+		namespace, err = findSandboxNamespace(n.client, ctx)
+		if err != nil {
+			return SandboxCredentials{}, err
+		}
+	}
+	token, err := getTokenForNamespace(n.client, ctx, namespace)
+	if err != nil {
+		return SandboxCredentials{}, err
+	}
+	return n.ResolveToken(ctx, token)
+}
+
+// getTokenForNamespace is a subroutine of ResolveNamespace wrapping the call to
+// obtain a JWT.  This step should eventually be eliminated in favor of an API
+// that goes directly from namespace to credentials.
+func getTokenForNamespace(client *godo.Client, ctx context.Context, namespace string) (string, error) {
+	path := "v2/function/namespaces/login_token"
+	body := postBody{Namespace: inputNamespace{Namespace: namespace}}
+	req, err := client.NewRequest(ctx, http.MethodPost, path, body)
+	if err != nil {
+		return "", err
+	}
+	tokenResponse := new(responseBody)
+	_, err = client.Do(ctx, req, tokenResponse)
+	if err != nil {
+		return "", err
+	}
+	return tokenResponse.Namespace.Token, nil
+}
+
+// findSandboxNamespace is a subroutine of ResolveNamespace implementing the search for
+// a sandbox namespace.
+func findSandboxNamespace(client *godo.Client, ctx context.Context) (string, error) {
+	path := "v2/function/namespaces"
+	req, err := client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return "", err
+	}
+	namespaces := new(namespaceList)
+	_, err = client.Do(ctx, req, namespaces)
+	if err != nil {
+		return "", err
+	}
+	sandboxes := []string{}
+	for _, ns := range namespaces.Namespaces {
+		if strings.Contains(ns.Label, "sandbox") {
```

Comment:
Not blocking for now, but I wonder how future proofed this is? Is a standalone offering, will customers be able to set their own labels with `sandbox` in them?

Reply:
Ah, just saw the comment about this being subject to change. :+1: